### DISCUSSION TITLE
Torch.solve is deprecated

### DISCUSSION THREAD

**Discussion by egelundsvej55** [kornia/kornia#1196] [Questions] [answered]
Hi

Thanking for writing this amazing library.

I am using it in google colab. When I do, I get the following message:

>  /usr/local/lib/python3.7/dist-packages/kornia/utils/helpers.py:96: UserWarning: torch.solve is deprecated in favor of torch.linalg.solveand will be removed in a future PyTorch release.
> torch.linalg.solve has its arguments reversed and does not return the LU factorization.
> To get the LU factorization see torch.lu, which can be used with torch.lu_solve or torch.lu_unpack.
> X = torch.solve(B, A).solution
> should be replaced with
> X = torch.linalg.solve(A, B) (Triggered internally at  /pytorch/aten/src/ATen/native/BatchLinearAlgebra.cpp:760.)
>   out1, out2 = torch.solve(input.to(dtype), A.to(dtype))

Is there someone who can tell me how to fix it so that the library will work even after they update PyTorch?

**Comment by shijianjian** [comment, marked as answer]
this is only a warning message and it should not block you from using it, right?

In the future versions, there are several solutions to work it around like a simple try-catch block, etc. No worries on that~!

**Comment by egelundsvej55** [comment]
Yeah its just a warning, I was just worried it would suddenly stop working altogether.. thanks 😊
